extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -51,9 +58,21 @@ void registerOpenTelemetryContextStorage(
 
     @BuildStep(onlyIf = OpenTelemetryEnabled.class)
     @Record(ExecutionTime.STATIC_INIT)
-    void createOpenTelemetry(OpenTelemetryRecorder recorder, Optional<TracerProviderBuildItem> tracerProviderBuildItem) {
+    void createOpenTelemetry(OpenTelemetryRecorder recorder, Optional<TracerProviderBuildItem> tracerProviderBuildItem,
+            LaunchModeBuildItem launchMode) {
+        if (launchMode.getLaunchMode() == LaunchMode.DEVELOPMENT) {
```

Comment:
I think this could go into a separate step with an `onlyIf=IsDevelopment`, plus the OT check. But not a big deal.

Reply:
I wanted to have them together so it was clear we were resetting the OT instance in dev mode before we set it again